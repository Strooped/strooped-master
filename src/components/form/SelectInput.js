import React from 'react';
import PropTypes from 'prop-types';

/**
 * Wrapper around <select/> with additional styles included
 * */
const SelectInput = (props) => {
  const {
    name,
    onBlur,
    onChange,
    value,
    label,
    children,
    ...other
  } = props;

  const inputId = `select-${name}`;

  return <div className="field">
    <label className="label has-text-light" htmlFor={inputId}>{label}</label>
    <div className="control">
      <div className="select is-fullwidth">
        <select
          name={name}
          id={inputId}
          onBlur={onBlur}
          onChange={onChange}
          value={value}
          {...other}
        >
          {children}
        </select>
      </div>
    </div>
  </div>;
};

SelectInput.propTypes = {
  name: PropTypes.string.isRequired,
  onBlur: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  value: PropTypes.any,
  children: PropTypes.node,
  label: PropTypes.string.isRequired,
};

export default SelectInput;
