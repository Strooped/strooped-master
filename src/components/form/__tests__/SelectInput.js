import React from 'react';
import { render, mount } from 'enzyme';
import SelectInput from '../SelectInput';

describe('<SelectInput/>', () => {
  it('should render all options', () => {
    const wrapper = render(<SelectInput
      onChange={() => null}
      onBlur={() => null}
      label="Some label"
      name="name"
    >
      <option value="1" disabled>Disabled option</option>
      <option value="2">Some option</option>
    </SelectInput>);

    expect(wrapper).toMatchSnapshot();
  });

  it('should notify when selection has changed', () => {
    const changeSpy = jest.fn();
    const wrapper = mount(<SelectInput
      onChange={changeSpy}
      onBlur={() => null}
      label="Some label"
      name="name"
      value="1"
    >
      <option value="1" disabled>Disabled option</option>
      <option value="2">Some option</option>
    </SelectInput>);

    const changeEvent = { target: { value: '2' } };
    wrapper.find('select').simulate('change', changeEvent);

    expect(changeSpy).toHaveBeenCalled();

    const [call] = changeSpy.mock.calls;
    expect(call[0].target.value).toBe('2');
  });
});
