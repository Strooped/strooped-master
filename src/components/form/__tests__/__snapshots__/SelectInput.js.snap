// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SelectInput/> should render all options 1`] = `
<div
  class="field"
>
  <label
    class="label has-text-light"
    for="select-name"
  >
    Some label
  </label>
  <div
    class="control"
  >
    <div
      class="select is-fullwidth"
    >
      <select
        id="select-name"
        name="name"
      >
        <option
          disabled=""
          value="1"
        >
          Disabled option
        </option>
        <option
          value="2"
        >
          Some option
        </option>
      </select>
    </div>
  </div>
</div>
`;
